pal/src/thread/thread.cpp

```diff
@@ -2788,21 +2788,37 @@ int CorUnix::CThreadMachExceptionHandlers::GetIndexOfHandler(exception_mask_t bm
 
 int GetCurrentThreadStackBounds(char** stackBase, char** stackEnd)
 {
+#ifdef _TARGET_MAC64
```

Comment:
TARGET_MAC64 is from PAL. Kept that as is. Maybe we should only use '__APPLE__'?